### ISSUE TITLE
several error in  PhyloEM

### ISSUE THREAD

**Issue by oscarIM** [pbastide/PhylogeneticEM#18]
Hello, This trying to use your program to estimate shifts in evolutionary optimal values of a single variable, over a relatively large phylogeny (ntips=1804), using the following code:

`PhyloEM(phylo = tree,Y_data = data,process = "OU",random.root = TRUE, stationary.root = TRUE,alpha = TRUE, parallel_alpha = TRUE, Ncores =40, progress.bar = TRUE, save_step = TRUE)`
However, it throws me a lot of error. The tree is ultrametric and binary, and the data without missing values (as one of the errors suggests)

Robust estimation of alpha by regression failed
Error in find_independent_regression_vectors.gglasso(Xkro, K, fit, root,  :Could not find K independent vectors in the regression path provided.

In lasso_regression_K_fixed.gglasso(Yvec = as.vector(Yp), Xkro = as.matrix(Xp),  :
  The solution fund by lasso had non independent vectors. Had to modify this solution.
Error in lasso_regression_K_fixed.gglasso(Yvec = as.vector(Yp), Xkro = as.matrix(Xp),  : 
  The selected variables do not produce a full rank regression matrix !
Error in gglasso::gglasso(x = Xp_orth, y = Yp_orth, group = group, loss = "ls",  : 
  Missing values in x not allowed!
Además: Warning message:
In init.EM.lasso(phylo = phylo, Y_data = Y_data, process = "OU",  :
  Lasso initialization fail : could not find a satisfying number of shifts. Proceeding to a default initialization.
Error in ls(bn, bs, ix, iy, gamma, nobs, nvars, x, y, pf, dfmax, pmax,  : 
  NA/NaN/Inf en llamada a una función externa (arg 9)
Además: Warning message:
In init.EM(phylo = phylo, Y_data = Y_data, Y_data_imp = Y_data_imp,  :
  Lasso initialization fail : could not find a satisfying number of shifts. Proceeding to a default initialization.
Error in lm.fit(x = Xp[, c(root, sh_ed), drop = FALSE], y = D) : 
  NA/NaN/Inf in 'y'
Además: Warning message:
In segmentation(phylo = phylo, nbr_of_shifts = nbr_of_shifts, conditional_law_X = conditional_law_X,  :
  At M step, Lasso regression failed.

Any advice? in advance, thank you very much

**Comment by pbastide** [comment]
Hi @oscarIM 

Thank you for your message, and for taking the time to try out PhyloEM.

The errors you see seem to indicate that the lasso initialization failed on your data. Unfortunately, the errors are not specific enough to allow me to point you in the right direction to solve your problem (sorry about that !).

From the other [issue](https://github.com/khabbazian/l1ou/issues/44) you opened a few days ago, it looks like `l1ou` is also failing on your data. So maybe there is something unusual in your data that make both methods fail ?

One way to debug this is to try and trim your tree to a smaller one, with less taxa, and see if the error still occurs. If you could come up with a **minimal example**, i.e. a tree as small as possible where the error still occurs, then I might be able to help you better.

Thanks again for your feedback, and I hope you'll be able to get meaningful results for your analysis.

**Comment by oscarIM** [comment]
Thank you for your prompt response. I'll do some tests and I'll write to you to tell you the results.

Thank you so much again.
